### PR TITLE
sg: add ability to define `checks` per `commandset`

### PR DESCRIPTION
**Pull request by mrnugget**
This allows us to check whether services are running before we boot up
the environment.

The changes to the config file format are backwards compatible. See the
`UnmarshalYAML` method on `Commandset`.

It also updates the checks to make sure that they work for users of the
docker-compose setup.

## Screenshots

Config:

screenshot_2021-07-20_12 41 14@2x

In action:

screenshot_2021-07-20_12 42 19@2x

### CODE REVIEW

File: dev/sg/config.go [hunk 1]
```diff
@@ -119,12 +128,38 @@ type Check struct {
...
 }
 
+type Commandset struct {
+	Name     string   `yaml:"-"`
+	Commands []string `yaml:"commands"`
+	Checks   []string `yaml:"checks"`
+}
+
+// UnmarshalYAML implements the Unmarshaler interface.
+func (c *Commandset) UnmarshalYAML(unmarshal func(interface{}) error) error {
```
Comment (eseliger) [inline dev/sg/config.go:138]: nice ✨

**Review by eseliger — APPROVED** [review]
I like it, and we had the postgres check in the bash script before, so I think we should add it. I also once ran into a problem when postgres wasn't running and everything was failing.

**Comment by tsenart** [comment]
Also like it, but could we go further and actually start those dependencies if they're needed?

**Comment by eseliger** [comment]
> Also like it, but could we go further and actually start those dependencies if they're needed?

That is probably very platform/installation dependent. I would want postgres to run when sg doesn't so I can run tests, for example. So it would probably need to start it as a daemon/service and there is probably no one-fits-all solution :/

**Comment by mrnugget** [comment]
> Also like it, but could we go further and actually start those dependencies if they're needed?

That's the next step 😎 

Vision: `sg setup` guides you through the setup of the dependencies, asks how you want to install Postgres and does so if it's not there, then persists the settings and uses them later to make sure everything's up and running.

That way a new teammember can do `curl -o /usr/local/bin/sg http://releases.github.com/sg.tar.gz` and type in `sg setup` to get started.

(A Stack Overflow engineer posted a screenshot of their tool installing all dependencies for them on Twitter a while back but I can't find it)

**Comment by mrnugget** [comment]
Confirmed with @fkling that the checks work with his docker-based setup, so I'm going to go ahead and merge this.

**Pull request merged**